src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOnSession.java

```diff
@@ -160,10 +161,14 @@ public void put(SecurityIdentity identity) {
                             connection.getInputStream().close();
                         } catch (Exception cause) {
                             log.warnHttpMechSsoFailedLogoutParticipant(remoteURI.toString(), cause);
+                            // failed to logout participant, remove it from the list of participants
+                            try (SingleSignOn target = context.getSingleSignOnManager().find(id)) {
+                                target.removeParticipant(participantId);
```

Comment:
I'm not sure. We are going to invalidate the SSO right after that given that participant list will always be empty.

The previous fix we did was related with making sure logout requests are send only from where the logout was triggered.